hyperspy/_signals/signal1d.py

```diff
@@ -1098,9 +1098,15 @@ def _remove_background_cli(
         else:
             try:
                 axis = self.axes_manager.signal_axes[0]
-                scale_factor = axis.scale if is_binned(self) else 1
+                if is_binned(self):
                 # in v2 replace by
-                # scale_factor = axis.scale if axis.is_binned else 1
+                # if axis.is_binned:
+                    if axis.is_uniform:
+                        scale_factor = axis.scale
+                    else:
+                        scale_factor = np.gradient(axis.axis)
```

Comment:
Similar as above, can be simplify to:
```suggestion
                    scale_factor = np.gradient(axis.axis)
```

Reply:
Similar as above. As here the single value scale is replaced by the whole vector gradient, I would leave it.